components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+        diags.dstnidep      = ekat::subview(diagnostic_scratch[33], icol);
+        diags.dstnicnt      = ekat::subview(diagnostic_scratch[34], icol);
+        diags.dstniimm      = ekat::subview(diagnostic_scratch[35], icol);
+        diags.bcnidep       = ekat::subview(diagnostic_scratch[36], icol);
+        diags.bcnicnt       = ekat::subview(diagnostic_scratch[37], icol);
+        diags.bcniimm       = ekat::subview(diagnostic_scratch[38], icol);
+        diags.numice10s     = ekat::subview(diagnostic_scratch[39], icol);
+        diags.numimm10sdst  = ekat::subview(diagnostic_scratch[40], icol);
+        diags.numimm10sbc   = ekat::subview(diagnostic_scratch[41], icol);
+        diags.stratiform_cloud_fraction =
+            ekat::subview(diagnostic_scratch[42], icol);
+
+        // assign cloud fraction
+        Kokkos::parallel_for(
+            Kokkos::TeamThreadRange(team, 0u, mam4::ndrop::pver),
+            [=](int klev) {
+              diags.stratiform_cloud_fraction(klev) =
+                  haero_atm.cloud_fraction(klev);
+            });
+        //-------------------------------------------------------------
+        // Heterogeneous freezing
+        // frzimm, frzcnt, frzdep are the outputs of
+        // hetfrz_classnuc_cam_calc used by the microphysics (e.g. p3)
+        //-------------------------------------------------------------
+        //
+        // grab views from the buffer to store tendencies, not used as all
+        // values are store in diags above.
+        const mam4::Tendencies tends(nlev);
+        const mam4::AeroConfig aero_config;
+        const Real t = 0;  // not used
+        hetfrz.compute_tendencies(aero_config, team, t, dt, haero_atm, surf,
+                                  progs, diags, tends);
+      });
+}
+}  // namespace
+
+MAMAci::MAMAci(const ekat::Comm &comm, const ekat::ParameterList &params)
+    : AtmosphereProcess(comm, params) {
+  // Asserts for the runtime or namelist options
+  EKAT_REQUIRE_MSG(m_params.isParameter("wsubmin"),
+                   "ERROR: wsubmin is missing from mam_aci parameter list.");
+  EKAT_REQUIRE_MSG(
+      m_params.isParameter("top_level_mam4xx"),
+      "ERROR: top_level_mam4xx is missing from mam_aci parameter list.");
+}
+
+// ================================================================
+//  SET_GRIDS
+// ================================================================
+void MAMAci::set_grids(
+    const std::shared_ptr<const GridsManager> grids_manager) {
+  // set grid for all the inputs and outputs
+  // use physics grid
+  grid_ = grids_manager->get_grid("Physics");
+
+  // Name of the grid
+  const auto &grid_name = grid_->name();
+
+  ncol_ = grid_->get_num_local_dofs();       // Number of columns on this rank
+  nlev_ = grid_->get_num_vertical_levels();  // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
```

Comment:
We may need to update this code before merging into master; see PR[2776](https://github.com/E3SM-Project/scream/pull/2776)

Reply:
I have already rebased the code and fixed these field layout tags.